assets/model_monitoring/components/src/shared_utilities/io_utils.py

```diff
@@ -177,34 +163,10 @@ def read_mltable_in_spark(mltable_path: str):
             raise InvalidInputError(f"Failed to read MLTable {mltable_path}, it is not from the same AML workspace.")
 
 
-def save_spark_df_as_mltable(metrics_df, folder_path: str, file_system=None):
+def save_spark_df_as_mltable(metrics_df, folder_path: str):
     """Save spark dataframe as mltable."""
-    # TODO: remove this explicit initialization after switch over to StoreUrl.
-    # init env-vars for supporting credential-less datastore scenarios.
-    # this env variable is required by Amlfs to get user token.
-    spark = init_spark()
-    spark_conf = spark.sparkContext.getConf()
-    spark_conf_vars = {
-        "AZUREML_SYNAPSE_CLUSTER_IDENTIFIER": "spark.synapse.clusteridentifier",
-        "AZUREML_SYNAPSE_TOKEN_SERVICE_ENDPOINT": "spark.tokenServiceEndpoint",
-    }
-    for env_key, conf_key in spark_conf_vars.items():
-        value = spark_conf.get(conf_key)
-        if value:
-            os.environ[env_key] = value
-
-    try:
-        metrics_df.write.mode("overwrite").parquet(folder_path)
-    except Exception as error:
-        # TODO: remove this check block after we switch over to using StoreUrl for saving spark df
-        if isinstance(error, Py4JJavaError):
-            if "Access token couldn't be obtained" in str(error):
-                raise InvalidInputError(
-                    MISSING_OBO_CREDENTIAL_HELPFUL_ERROR_MESSAGE.format(message=error.java_exception.getMessage()))
-        raise error
-
     base_path = folder_path.rstrip('/')
-    output_path_pattern = base_path + "/*.parquet"
+    output_path_pattern = base_path + "/data/*.parquet"
```

Comment:
why add /data sub folder?

Reply:
Because when I did the MLTable write first (to enable credential-less with StoreUrl) and then did `spark.write.parquet` in the same location it ended up overwriting and deleting the MLTable, so if we write to the data subfolder we avoid this issue.